src/Migrations/M049MigrateCoversBlockToActionsListBlock.php

```diff
@@ -0,0 +1,288 @@
+     */
+    private static function transform_block(array $block): array
+    {
+        $existing_block_attrs = self::get_posts_list_block_attrs($block);
+
+        $posts_override = $existing_block_attrs['posts'];
+        $layout_type = $existing_block_attrs['layout'] === 'carousel' ? 'flex' : 'grid';
+        $classname = Utils\Constants::ACTIONS_LIST . ' p4-query-loop is-custom-layout-' . $existing_block_attrs['layout'];
+        $per_page = $existing_block_attrs['per_page'];
+        $current_post_id = $existing_block_attrs['current_post_id'];
+        $additional_class = $existing_block_attrs['additional_class'];
+
+        $attrs = self::set_query_block_attrs($posts_override, $layout_type, $per_page, $current_post_id, $classname);
+
+        $inner_blocks = [
+            self::get_head_block($existing_block_attrs['title']),
+            self::get_paragraph_block($existing_block_attrs['description']),
+            self::get_query_no_results_block(),
+            self::get_post_template(),
+            self::get_buttons_block(),
+        ];
+
+        return Utils\Functions::create_block_query(
+            $inner_blocks,
+            $attrs,
+            $additional_class ? $classname . ' ' . $additional_class : $classname,
+            Utils\Constants::ACTIONS_LIST
+        );
+    }
+
+    /**
+     * Get the attributes from a Covers block.
+     */
+    private static function get_posts_list_block_attrs(array $existing_block): array
+    {
+        $attrs = $existing_block['attrs'];
+
+        return [
+            'title' => isset($attrs['title']) ? $attrs['title'] : '',
+            'description' => isset($attrs['description']) ? $attrs['description'] : '',
+            'posts' => isset($attrs['posts']) ? $attrs['posts'] : [],
+            'post_types' => isset($attrs['post_types']) ? $attrs['post_types'] : [],
+            'current_post_id' => isset($attrs['current_post_id']) ? $attrs['current_post_id'] : 0,
+            'layout' => isset($attrs['layout']) ? $attrs['layout'] : 'grid',
+            'per_page' => isset($attrs['initialRowsLimit']) ? $attrs['initialRowsLimit'] : 1,
+            'additional_class' => isset($attrs['additional_class']) ? $attrs['additional_class'] : '',
+        ];
+    }
+
+    /**
+     * Set all the attributes to create a new Query block.
+     */
+    private static function set_query_block_attrs(
+        array $posts_override,
+        string $layout_type,
+        int $per_page,
+        int $current_post_id,
+        string $classname,
+    ): array {
+
+        if ($per_page === 1) {
+            $items_per_page = 3;
+        } elseif ($per_page === 2) {
+            $items_per_page = 6;
+        } else {
+            $items_per_page = 100;
+        }
+
+        // If the layout is carousel (flex), we need at least 4 items per page.
+        if ($layout_type === 'flex' && $items_per_page < 4) {
+            $items_per_page = 4;
+        }
+
+        $query = [];
```

Comment:
I think this is more cleaner. What do you think?
```suggestion
        $query = [
            'pages' => 0,
            'perPage' => $items_per_page,
            'offset' => 0,
            'author' => '',
            'search' => '',
            'exclude' => [$current_post_id],
            'sticky' => '',
            'inherit' => false,
            'postType' => Utils\Constants::POST_TYPES_ACTION,
            'postIn' => $posts_override,
            'hasPassword' => false,
            'order' => 'desc',
            'orderBy' => 'date',
        ];
```

Reply:
Done, thanks!